src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -215,6 +216,199 @@ public bool Read()
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <remarks>
+        /// If the look up text is invalid UTF-8 text, the method will always return false since you cannot have 
+        /// invalid UTF-8 within the JSON payload.
+        /// </remarks>
+        /// <remarks>
+        /// The comparison of the JSON token value in the source and the look up text is done by first unescaping the JSON value in source,
+        /// if required. The look up text is matched as is, without any modifications to it.
+        /// </remarks>
+        public bool TextEquals(ReadOnlySpan<byte> otherUtf8Text)
+        {
+            if (TokenType != JsonTokenType.PropertyName && TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedStringComparison(TokenType);
+            }
+
+            if (HasValueSequence)
+            {
+                return CompareToSequence(otherUtf8Text);
+            }
+
+            if (_stringHasEscaping)
+            {
+                return UnescapeAndCompare(otherUtf8Text);
+            }
+
+            return otherUtf8Text.SequenceEqual(ValueSpan);
+        }
+
+        /// <summary>
+        /// Compares the UTF-16 encoded text to the unescaped JSON token value in the source and returns true if they match.
+        /// </summary>
+        /// <param name="otherText">The UTF-16 encoded text to compare against.</param>
+        /// <returns>True if the JSON token value in the source matches the UTF-16 encoded look up text.</returns>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to find a text match on a JSON token that is not a string
+        /// (i.e. other than <see cref="JsonTokenType.String"/> or <see cref="JsonTokenType.PropertyName"/>).
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        /// <remarks>
+        /// If the look up text is invalid or incomplete UTF-16 text (i.e. unpaired surrogates), the method will always return false
+        /// since you cannot have invalid UTF-16 within the JSON payload.
+        /// </remarks>
+        /// <remarks>
+        /// The comparison of the JSON token value in the source and the look up text is done by first unescaping the JSON value in source,
+        /// if required. The look up text is matched as is, without any modifications to it.
+        /// </remarks>
+        public bool TextEquals(ReadOnlySpan<char> otherText)
+        {
+            if (TokenType != JsonTokenType.PropertyName && TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedStringComparison(TokenType);
+            }
+
+            byte[] otherUtf8TextArray = null;
+
+            Span<byte> otherUtf8Text;
+
+            // Transcoding has a max factor of 3, but calling AsBytes increases it by x2
```

Comment:
Seems like this comment should be on the declaration of MaxExpansionFactorWhileEscaping rather than here.  The 3 and x2 have no meaning at this point, since there's no in the subsequent source.

Reply:
I also looked up the constant to figure out where 3 and 2 were coming from or going.